core/src/main/java/org/pentaho/di/cluster/SlaveConnectionManager.java

```diff
@@ -128,6 +130,7 @@ private static X509TrustManager getDefaultTrustManager() {
     return new X509TrustManager() {
       @Override
       public void checkClientTrusted( X509Certificate[] certs, String param ) throws CertificateException {
+        throw new CertificateException( "Client Authentication not implemented" );
```

Comment:
I'm not familiar with how we're managing SSL here.  Does this method get called by anything that was relying on it being a noop?

Reply:
It should not be getting called from anywhere.  It's used during client SSL authentication.  The only usage is in initiating a connection to the server.  Having the exception in place stops us from unknowingly extending the usage of this to a client SSL situation where it wouldn't be secured.